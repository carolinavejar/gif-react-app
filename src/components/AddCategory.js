import React, { useState } from 'react'

export const AddCategory = () => {
    const  [inputValue, setInputValue] = useState('Hola');

    const handleInputChange = (e)=> {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e)=> {
        e.preventDefault();
        console.log(` * * *    handleSubmit    * * * `);;
    }
    return (
        <form onSubmit = { handleSubmit }>
            <input 
                type = "text"
                value =  { inputValue }
                className = "input-field"
                onChange = { handleInputChange }
            />
        </form>
    )
}
