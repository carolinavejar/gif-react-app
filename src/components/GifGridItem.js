import React from 'react'
import { shallow } from 'enzyme/build';

import Enzyme from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';

export const GifGridItem = ( {id, title, url} ) => {
    return (
        <div className = "card animate__animated animate__bounce animate__delay-2s">
            <div className = "card-image">
                <img src= { url} alt = {title}></img> 
                <div className = "card-title">{title} </div>
            </div>
        </div>
    )
}
