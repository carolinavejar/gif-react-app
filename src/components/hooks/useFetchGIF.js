import { useState} from 'react'

export const useFetchGIF = () => {
    const [state, setstate] = useState({
        data: [],
        loading: true
    });

    setTimeout(() => {
        setstate({
            data: [11,12],
            loading: false
        })
    }, 3000);

    return state;  
    //  state 
    // ========
    // {
    //     data: [],
    //     loading: true
    // }
}
