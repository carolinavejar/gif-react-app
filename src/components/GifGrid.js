import React , { useState, useEffect } from 'react';
import axios from 'axios';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, []);
    // ejecuta el cÃ³digo de arriba solo la primera vez que se renderiza

    const getGifs = () => {
        const api_key = 'URLjL49KyzdscasPa5146vq3CmpIYQjq';
        const cat = 'RIHANNA';
        axios.get(`https://api.giphy.com/v1/gifs/search?api_key=${ api_key }&q=${ cat }&limit=10`)
        .then((response) => {
            const { data : { data } } = response;
            const gifs = data.map ( img => {
                return {
                    id: img.id,
                    title: img.title,
                    url: img.images.downsized_medium.url
                }
            });
            setImages(gifs);
        })
    };

    return (
        <>
            {
                images.map( (img) => 
                    <GifGridItem 
                        key = { img.id }
                        { ...img }
                    >

                    </GifGridItem>
                )
            }
        </>
    )
}
