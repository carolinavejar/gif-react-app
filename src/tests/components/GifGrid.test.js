import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGIF } from '../../components/hooks/useFetchGIF';
jest.mock('../../components/hooks/useFetchGIF');
describe('Pruebas componente GifGrid', () =>{
    const cat = "Rihanna";
    // const wrapper = shallow(<GifGrid category = {cat}  />);

    test('debe hacer match con snapshop', () => {
        useFetchGIF.mockReturnValue({
            data : [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category = {cat}  />);
        expect (wrapper).toMatchSnapshot();
    });

    test('debe mostrar cuando se cargan imagenes', () => {
        const imgs = [{
            id: '1',
            url: 'https://url.del.gif',
            title: 'Titulo GIF'
        }]
        useFetchGIF.mockReturnValue({
            data : imgs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category = {cat}  />);
        expect (wrapper).toMatchSnapshot();
    });
})
