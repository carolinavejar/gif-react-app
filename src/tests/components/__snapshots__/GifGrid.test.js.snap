// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente GifGrid debe hacer match con snapshop 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Rihanna
     
  </h3>
  <p>
     Loading ...
  </p>
</Fragment>
`;

exports[`Pruebas componente GifGrid debe mostrar cuando se cargan imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Rihanna
     
  </h3>
  <GifGridItem
    id="1"
    key="1"
    title="Titulo GIF"
    url="https://url.del.gif"
  />
</Fragment>
`;
